python/ray/monitor.py

```diff
@@ -93,24 +95,36 @@ def parse_resource_demands(self, resource_load_by_shape):
         """Handle the message.resource_load_by_shape protobuf for the demand
         based autoscaling. Catch and log all exceptions so this doesn't
         interfere with the utilization based autoscaler until we're confident
-        this is stable.
+        this is stable. Worker queue backlogs are added to the appropriate
+        resource demand vector.
 
         Args:
             resource_load_by_shape (pb2.gcs.ResourceLoad): The resource demands
                 in protobuf form or None.
+
+        Returns:
+            List[ResourceDict]: Waiting bundles (ready and feasible).
+            List[ResourceDict]: Infeasible bundles.
         """
         waiting_bundles, infeasible_bundles = [], []
         try:
-            if self.autoscaler:
-                for resource_demand_pb in list(
-                        resource_load_by_shape.resource_demands):
-                    request_shape = dict(resource_demand_pb.shape)
-                    for _ in range(
-                            resource_demand_pb.num_ready_requests_queued):
-                        waiting_bundles.append(request_shape)
-                    for _ in range(
-                            resource_demand_pb.num_infeasible_requests_queued):
+            for resource_demand_pb in list(
```

Comment:
Why remove the self.autoscaler check?

Reply:
It wasn't really useful (I think I was being overly defensive when I added it to begin with) and was getting in the way of writing a unit test.